src/rlang-dev.c

```diff
@@ -1,8 +1,19 @@
 #include "vctrs.h"
+#include "decl/rlang-dev-decl.h"
 
 const char* r_friendly_type_of(r_obj* x) {
-  r_obj* call = KEEP(r_parse("friendly_type_of(x)"));
-  r_obj* ffi_out = KEEP(r_eval_with_x(call, x, vctrs_ns_env));
+  return r_friendly_type_of_opts(x, true, false);
+}
+
+const char* r_friendly_type_of_length(r_obj* x) {
+  return r_friendly_type_of_opts(x, true, true);
+}
+
+const char* r_friendly_type_of_opts(r_obj* x,
+                                    bool value,
+                                    bool length) {
+  r_obj* call = KEEP(r_parse("friendly_type_of(x, value = y, length = z)"));
+  r_obj* ffi_out = KEEP(r_eval_with_xyz(call, x, r_lgl(value), r_lgl(length), vctrs_ns_env));
```

Comment:
Do the `r_lgl()`s need protection?

Reply:
Is this because `r_lgl()` uses `R_ScalarLogical()`, which doesn't allocate at all?

I didn't know this was the definition of that, which is neat https://github.com/wch/r-source/blob/79298c499218846d14500255efd622b5021c10ec/src/include/Rinlinedfuns.h#L992-L998